### ISSUE TITLE
Nice to have: Including custom headers on completion/chat

### ISSUE THREAD

**Issue by ScotterC** [alexrudall/ruby-openai#266]
I was looking into using [Helicone.ai's caching feature](https://docs.helicone.ai/advanced-usage/caching) but it requires including a header in the openai request. Here's what it looks like in the python lib:

```python
openai.api_base = "https://oai.hconeai.com/v1"

openai.Completion.create(
    model="text-davinci-003",
    prompt="How do I enable caching?",
    headers={
      "Helicone-Cache-Enabled": "true",
    }
)
```

For ruby-openai it could look like

```ruby
response = client.completions(
    parameters: {
        model: "text-davinci-001",
        prompt: "Once upon a time",
        max_tokens: 5,
        headers: {
          "Helicone-Cache-Enabled": "true",
        }
    })
```

**Comment by ch0ngxian** [comment]
Already have someone open PR for this enhancement.
https://github.com/alexrudall/ruby-openai/pull/257

**Comment by ScotterC** [comment]
Right on. Nice catch.

**Comment by alexrudall** [comment]
Released in [v4.3](https://rubygems.org/gems/ruby-openai/versions/4.3.0) - See [README](https://github.com/alexrudall/ruby-openai#custom-timeout-or-base-uri)
